aspnet/mvc/models/file-uploads.md

```diff
@@ -0,0 +1,455 @@
+
+![image](file-uploads/_static/upload-form.png)
+
+The individual files uploaded to the server can be accessed through [Model Binding](model-binding.md) using the [IFormFile](http://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Http/IFormFile/index.html.md#Microsoft.AspNetCore.Http.IFormFile.md) interface. `IFormFile` has the following structure:
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+````c#
+
+   public interface IFormFile
+   {
+     string ContentType { get; }
+     string ContentDisposition { get; }
+     IHeaderDictionary Headers { get; }
+     long Length { get; }
+     string Name { get; }
+     string FileName { get; }
+     Stream OpenReadStream();
+     void CopyTo(Stream target);
+     Task CopyToAsync(Stream target, CancellationToken cancellationToken = null);
+   }
+   ````
+
+When uploading files using model binding and the `IFormFile` interface, the action method can accept either a single `IFormFile` or an `IEnumerable<IFormFile>` (or `List<IFormFile>`) representing several files. The following example loops through one or more uploaded files, saves them to the local file system, and then returns the total number and size of files uploaded.
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/mvc/models/file-uploads/sample/FileUploadSample/Controllers/UploadFilesController.cs", "highlight_args": {"linenostart": 1, "hl_lines": [21, 27, 34]}, "names": []} -->
+
+````c#
+
+   using System.Collections.Generic;
+   using System.IO;
+   using System.Linq;
+   using System.Threading.Tasks;
+   using Microsoft.AspNetCore.Hosting;
+   using Microsoft.AspNetCore.Http;
+   using Microsoft.AspNetCore.Mvc;
+
+   namespace FileUploadSample.Controllers
+   {
+       public class UploadFilesController : Controller
+       {
+           private readonly IHostingEnvironment _hostingEnvironment;
+
+           public UploadFilesController(IHostingEnvironment hostingEnvironment)
+           {
+               _hostingEnvironment = hostingEnvironment;
+           }
+
+           [HttpPost("UploadFiles")]
+           public async Task<IActionResult> Post(List<IFormFile> files)
+           {
+               long size = files.Sum(f => f.Length);
+
+               // save files - "uploads" path must already exist
+               var uploadPath = Path.Combine(_hostingEnvironment.WebRootPath, "uploads");
+               foreach (var formFile in files)
+               {
+                   if (formFile.Length > 0)
+                   {
+                       var filePath = Path.Combine(uploadPath, formFile.FileName);
```

Comment:
I assume we want to distrust the filename just like any other user-submitted content. To that end, it maybe makes sense to sanitize the filename or just ignore it and use a convention of some kind to generate the names. However, any additional guidance on how FileName should be used (if at all - and if not why is it there?) would be appreciated.